contracts/Scroll_SpokePool.sol

```diff
@@ -4,6 +4,12 @@ pragma solidity ^0.8.0;
 import "./SpokePool.sol";
 import "@scroll-tech/contracts/L2/gateways/IL2GatewayRouter.sol";
 import "@scroll-tech/contracts/libraries/IScrollMessenger.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+interface IL2GatewayRouterExtended is IL2GatewayRouter {
```

Comment:
Nit: prefer naming this like IL2ERC20GatewayRouter

Reply:
I'm hesitant to use `IL2ERC20GatewayRouter` since `IL2ERC20Gateway` is already defined by the Scroll ABI and is inherited by `IL2GatewayRouter`, so the concern was that the naming ends up extremely similar and it becomes easy to mix them up. Is there another name we could go for?

Reply:
I think its fine